app.py

```diff
@@ -60,35 +83,15 @@ def sync(force_sync: bool, graceful: bool, debug: bool, document_type: Optional[
     if document_type:
         # We sync only a specific category of Thoth documents
         function = HANDLERS_MAPPING[document_type]
-        stats = function(force=force_sync, graceful=graceful, graph=graph)
+        to_sync = [(document_type, function)]
+    else:
+        to_sync = thoth_sync_module.__dict__.items()
```

Comment:
function fetching from the file module seems like it would include vars and additional bits as well.
we just want the type and function names, as stated in HANDLERS_MAPPING, maybe we can reuse it ?

also, our use case is, 
if no for particular document are not sync, then trigger them with explicity call 
or else only sync solver , as that is having the data ingestion knowledge

with this flow, if the document type is not found it would try to sync all type 
which could be time-consuming.

Reply:
> function fetching from the file module seems like it would include vars and additional bits as well. we just want the type and function names, as stated in HANDLERS_MAPPING, maybe we can reuse it ?

True. We do filter (as the previous code did, I merely changed the place where this is used) on `sync_`. I'm not opposed to the change, but the behaviour is not strictly equivalent (there is a `sync_documents` at the end of the sync.py thoth/storages module.

Actually, that might be a (pre-existing) bug.

> 
> also, our use case is, if no for particular document are not sync, then trigger them with explicity call or else only sync solver , as that is having the data ingestion knowledge
> 
> with this flow, if the document type is not found it would try to sync all type which could be time-consuming.

I'm not sure what you mean. The only thing that would change in the control flow is that solver sync is not performed first , is that important somehow ? The rest is only 'having one execution loop and determine the items in the loop before hands' instead of 'determine several executions items'

Reply:
> if no for particular document are not sync, then trigger them with explicity call or else only sync solver , as that is having the data ingestion knowledge
> 
> with this flow, if the document type is not found it would try to sync all type which could be time-consuming.


The code on master does not read that way. There is and explicit `return` on the `if document_type` case, but not after sync_solver_documents call. So the flow on master would be:

1. document type is provided -> only that type is synced
2. Not provided -> solver are synced, then all types are synced, skipping solver.

If the intended behavior is to only sync one type at a time, we should remove the Optional on document_type (and also, why is that loop doing there in that case ?)

Reply:
You are correct.
Thanks for mentioning this in detail.

about the loop, it seems like a debug statement left from initial write of the code.